shared_vae_class.py

```diff
@@ -67,250 +83,200 @@ def sampling(args):
-        # divergence
-        def vae_loss(inputs, finalLayer):
-            reconstruction_loss = K.sum(K.square(finalLayer - inputs))
-            kl_loss = - 0.5 * K.sum(1 + z_log_sigmaFull - K.square(
-                z_meanFull) - K.square(K.exp(z_log_sigmaFull)), axis=-1)
-            total_loss = K.mean(reconstruction_loss + kl_loss)
-            return total_loss
-
         # Loss function for the left VAE
         # Loss function comprised of two parts, Cross_entropy, and
         # divergence
         def left_vae_loss(inputs, finalLayer):
+            # Other options for loss function:
+            # reconstruction_loss = self.params.inputSizeLeft *
+            #                       binary_crossentropy(finalLayer, inputs)
+            # reconstruction_loss = self.params.inputSizeLeft *
+            #                       mse(finalLayer, inputs)
             reconstruction_loss = K.sum(K.square(finalLayer - inputs))
             kl_loss = - 0.5 * K.sum(1 + z_log_sigmaLeft - K.square(
-                z_meanLeft) - K.square(K.exp(z_log_sigmaLeft)), axis=-1)
-            total_loss = K.mean(reconstruction_loss + kl_loss)
+                z_meanLeft) - K.exp(z_log_sigmaLeft), axis=-1)
+            total_loss = K.mean(reconstruction_loss +
+                                (K.get_value(self.betaLeft) * kl_loss))
             return total_loss
 
         # Loss function for the right VAE
         # Loss function comprised of two parts, Cross_entropy, and
         # divergence
         def right_vae_loss(inputs, finalLayer):
+            # Other options for loss function:
+            # reconstruction_loss = self.params.inputSizeRight *
+            #                       binary_crossentropy(finalLayer, inputs)
+            # reconstruction_loss = self.params.inputSizeRight *
+            #                       mse(finalLayer, inputs)
             reconstruction_loss = K.sum(K.square(finalLayer - inputs))
             kl_loss = - 0.5 * K.sum(1 + z_log_sigmaRight - K.square(
-                z_meanRight) - K.square(K.exp(z_log_sigmaRight)), axis=-1)
-            total_loss = K.mean(reconstruction_loss + kl_loss)
+                z_meanRight) - K.exp(z_log_sigmaRight), axis=-1)
+            total_loss = K.mean(reconstruction_loss +
+                                (K.get_value(self.betaRight) * kl_loss))
             return total_loss
 
         # Define the Encoder with the Left and Right branches
         leftEncoderInput = Input(shape=(self.params.inputSizeLeft,))
+        # leftEncoderInputNorm = BatchNormalization()(leftEncoderInput)
         leftEncoderFirstLayer = Dense(
             self.params.firstLayerSizeLeft,
             activation='relu')(leftEncoderInput)
+        # leftEncoderFirstLayerNorm = BatchNormalization()
+        #                             (leftEncoderFirstLayer)
+        leftEncoderFirstLayerDropOut = Dropout(
+            self.params.dropout)(leftEncoderFirstLayer)
+
+        '''
```

Comment:
Why a string here?

Reply:
This is the second layer which I have chosen not to use, but I want to leave the code in so that it is easy to re-implement.